### DISCUSSION TITLE
Structure generation from cation defect spinel

### DISCUSSION THREAD

**Discussion by roentgenium** [giprayogo/SHRY#12] [Q&A] [answered]
Dear all,

I'm new to SHRY and trying to generate structures from a possible structure of cation defect spinel, gamma-alumina.
I try to use the cif file below (original file), but had an error message, "Composition is not rational!". How could I resolve this?
I changed occupancy as "modified file 1" (below), then I successfully obtained 22 slices, so it works, but actual occupancy should be the original one.

Original file is like this.
> data
> _chemical_name_common                  'g-alumina'
> _cell_length_a                         7.911000
> _cell_length_b                         7.911000
> _cell_length_c                         7.911000
> _cell_angle_alpha                      90.000000
> _cell_angle_beta                       90.000000
> _cell_angle_gamma                      90.000000
> _cell_volume                           495.101357
> _space_group_name_H-M_alt              'F d -3 m'
> _space_group_IT_number                 227
> 
> loop_
> _space_group_symop_operation_xyz
> (snip)
> 
> loop_
>    _atom_site_label
>    _atom_site_occupancy
>    _atom_site_fract_x
>    _atom_site_fract_y
>    _atom_site_fract_z
>    _atom_site_adp_type
>    _atom_site_B_iso_or_equiv
>    _atom_site_type_symbol
>    Al1        0.8333  0.000000     0.000000     0.000000    Biso  0.500000 Al
>    Al2        1.0     0.375000     0.375000     0.375000    Biso  0.500000 Al
>    O1         1.0     0.245350     0.245350     0.245350    Biso  1.000000 O
>    X1         0.1667  0.000000     0.000000     0.000000    Biso  0.500000 X

Modified file 1 is like this (modified last section).
> (snip)
>    Al1        0.75 0.000000     0.000000     0.000000    Biso  0.500000 Al
>    Al2        1.0     0.375000     0.375000     0.375000    Biso  0.500000 Al
>    O1         1.0     0.245350     0.245350     0.245350    Biso  1.000000 O
>    X1         0.25  0.000000     0.000000     0.000000    Biso  0.500000 X

**Comment by giprayogo** [comment, marked as answer]
Hi, Thanks for reporting the issue!

One workaround is by adding more precision, e.g:
> Al1 0.8333333 0.000000 0.000000 0.000000 Biso 0.500000 Al
> Al2 1.0 0.375000 0.375000 0.375000 Biso 0.500000 Al
> O1 1.0 0.245350 0.245350 0.245350 Biso 1.000000 O
> X1 0.1666667 0.000000 0.000000 0.000000 Biso 0.500000 X

**Reply by roentgenium** [reply]
Thank you for your prompt response! It works!
I'm not so familiar with programming , but I guess it is related to handling float in Python.
Next, I got a message, "Can't fit Al5X1 within 16 sites", and then I put an option "-s 3 1 1".
It completely worked well and told me "Expected unique patterns is 1972317" with  "--count-only" option.
So, it's time to think.
Anyway, I would like to thank again for sharing this code!

**Reply by kousuke-nakano** [reply]
Dear @roentgenium, thanks for your post! 

> Next, I got a message, "Can't fit Al5X1 within 16 sites", and then I put an option "-s 3 1 1".
It completely worked well and told me "Expected unique patterns is 1972317" with "--count-only" option.
So, it's time to think.

Would you give us your CIF file, if possible, to pinpoint the problem? Thanks.

**Reply by roentgenium** [reply]
Here is the original cif file.

[g-Al2O3_66559_modified.txt](https://github.com/giprayogo/SHRY/files/12714148/g-Al2O3_66559_modified.txt)

If I understand it correctly, SHRY works perfectly after adding more precision.
SHRY simply gave many (exact, I guess) possible configurations.
At present, I chose one as an initial model for MD simulation. So, no problem!

**Reply by kousuke-nakano** [reply]
Hi @roentgenium 

> If I understand it correctly, SHRY works perfectly after adding more precision.

Yes.

> SHRY simply gave many (exact, I guess) possible configurations.

Yes, exactly. I have also confirmed that the pull-request #13 works for the attached CIF file without adding many precisions. `shry g-Al2O3_66559_x.cif --count-only -s 3 1 1`. We will merge it soon to the main.

[g-Al2O3_66559_x.txt](https://github.com/giprayogo/SHRY/files/12722282/g-Al2O3_66559_x.txt)

> At present, I chose one as an initial model for MD simulation. So, no problem!

Perfect! 

Please feel free to post another discussion if there is unclear stuff, and we would appreciate it if you would cite our paper to advertise SHRY when publishing a paper.

**Comment by kousuke-nakano** [comment]
Dear @giprayogo, Hi! This is a frequently asked question, isn't it? How about adding some warning about it? Indeed, if SHRY detects that someone seems to use 1/3, 1/5, etc...., we can let SHRY print some warning or suggestion, such as "Plz. add more precision, etc.".

**Comment by kousuke-nakano** [comment]
Dear @giprayogo and @roentgenium, I have fixed the bug in the pull request #13.
